src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.Unix.cs

```diff
@@ -44,18 +44,60 @@ public static void VerifyPackageInfo()
         {
-            return EncryptDecryptHelper(securityContext, input, offset: 0, size: 0, encrypt: true, output: ref output, resultSize: out resultSize);
+            resultSize = 0;
+            try
+            {
+                Interop.Ssl.SslErrorCode errorCode = Interop.Ssl.SslErrorCode.SSL_ERROR_NONE;
+
+                resultSize = Interop.OpenSsl.Encrypt(securityContext.SslContext, input.Span, ref output, out errorCode);
+
+                return MapNativeErrorCode(errorCode);
+            }
+            catch (Exception ex)
+            {
+                return new SecurityStatusPal(SecurityStatusPalErrorCode.InternalError, ex);
+            }
         }
 
         public static SecurityStatusPal DecryptMessage(SafeDeleteSslContext securityContext, byte[] buffer, ref int offset, ref int count)
         {
-            SecurityStatusPal retVal = EncryptDecryptHelper(securityContext, buffer, offset, count, false, ref buffer, out int resultSize);
-            if (retVal.ErrorCode == SecurityStatusPalErrorCode.OK ||
-                retVal.ErrorCode == SecurityStatusPalErrorCode.Renegotiate)
+            int resultSize;
+            try
+            {
+                Interop.Ssl.SslErrorCode errorCode = Interop.Ssl.SslErrorCode.SSL_ERROR_NONE;
+
+                resultSize = Interop.OpenSsl.Decrypt(securityContext.SslContext, new Span<byte>(buffer, offset, count), out errorCode);
+
+                SecurityStatusPal retVal = MapNativeErrorCode(errorCode);
+
+                if (retVal.ErrorCode == SecurityStatusPalErrorCode.OK ||
+                    retVal.ErrorCode == SecurityStatusPalErrorCode.Renegotiate)
+                {
+                    count = resultSize;
+                }
+
+                return retVal;
+            }
+            catch (Exception ex)
+            {
+                return new SecurityStatusPal(SecurityStatusPalErrorCode.InternalError, ex);
+            }
+        }
+
+        private static SecurityStatusPal MapNativeErrorCode(Interop.Ssl.SslErrorCode errorCode)
+        {
+            switch (errorCode)
             {
-                count = resultSize;
+                case Interop.Ssl.SslErrorCode.SSL_ERROR_RENEGOTIATE:
+                    return new SecurityStatusPal(SecurityStatusPalErrorCode.Renegotiate);
+                case Interop.Ssl.SslErrorCode.SSL_ERROR_ZERO_RETURN:
+                    return new SecurityStatusPal(SecurityStatusPalErrorCode.ContextExpired);
+                case Interop.Ssl.SslErrorCode.SSL_ERROR_NONE:
+                case Interop.Ssl.SslErrorCode.SSL_ERROR_WANT_READ:
+                    return new SecurityStatusPal(SecurityStatusPalErrorCode.OK);
+                default:
+                    return new SecurityStatusPal(SecurityStatusPalErrorCode.InternalError, new Interop.OpenSsl.SslException((int)errorCode));
             }
-            return retVal;
         }
```

Comment:
Nit: might be slightly more succient as a switch expression?
```C#
private static SecurityStatusPal MapNativeErrorCode(Interop.Ssl.SslErrorCode errorCode) =>
    errorCode switch
    {
        Interop.Ssl.SslErrorCode.SSL_ERROR_RENEGOTIATE => new SecurityStatusPal(SecurityStatusPalErrorCode.Renegotiate),
        Interop.Ssl.SslErrorCode.SSL_ERROR_ZERO_RETURN => new SecurityStatusPal(SecurityStatusPalErrorCode.ContextExpired),
        Interop.Ssl.SslErrorCode.SSL_ERROR_NONE or Interop.Ssl.SslErrorCode.SSL_ERROR_WANT_READ => new SecurityStatusPal(SecurityStatusPalErrorCode.OK),
        _ => new SecurityStatusPal(SecurityStatusPalErrorCode.InternalError, new Interop.OpenSsl.SslException((int)errorCode)),
    };
```

Reply:
done